include/nuttx/can/error.h

```diff
@@ -0,0 +1,127 @@
+/* SPDX-License-Identifier: ((GPL-2.0-only WITH Linux-syscall-note) OR BSD-3-Clause) */
```

Comment:
So this header is borrowed from Linux and created by Oliver Hartkopp and is BSD or GPL licensed. @patacongo for the initial commits in this SocketCAN you've also reused some code Linux, the question what's the correct to change this header to NuttX code format and Apache license yet still reference to the original BSD licensed material?

Reply:
> So this header is borrowed from Linux and created by Oliver Hartkopp and is BSD or GPL licensed. @patacongo for the initial commits in this SocketCAN you've also reused some code Linux, the question what's the correct to change this header to NuttX code format and Apache license yet still reference to the original BSD licensed material?

That is beyond my knowledge of the intricacies of licensing.  @justinmclean should have some input.  We did discuss this strange dual license. It is from the original VW SocketCAN code.  At that time, the advice was this this should not be an obstacle.

I don't think we should change anything until we have a good understanding.  I believe that we can merge the code onto a branch and make sure that we have what even license headers, ICLAs, or software grants in place before we merge to master.

Reply:
It's BSD licensed so that fine. The BSD license need to be added to our LICENSE file (if it hasn't been). Unless major changes have been made I would leave the original BSD headers on the code and not put an ASF header on it.